test/gui/shared/steps/steps.py

```diff
@@ -362,4 +369,16 @@ def step(context, receiver, resource):
 def step(context, fileShareContext):
     test.compare(str(waitForObjectExists(names.sharingDialog_The_file_can_not_be_shared_because_it_was_shared_without_sharing_permission_QLabel).text), fileShareContext)
 
-                 
\ No newline at end of file
+ 
+@When('the user creates a new public link for file "|any|" without password using the client-UI')
+def step(context, resource):
+    test.compare(str(waitForObjectExists(names.sharingDialog_label_name_QLabel).text), resource)
+    clickButton(waitForObject(names.oCC_ShareLinkWidget_createShareButton_QPushButton))     
+    
+@When('the user creates a new public link for file "|any|" with password "|any|" using the client-UI')
+def step(context, resource, password):
+    test.compare(str(waitForObjectExists(names.sharingDialog_label_name_QLabel).text), resource)
+    snooze(10)
```

Comment:
why is the sleep here?

Reply:
That was added to get object for password field but we won't be needing it now. I will remove it in my next commit.